### ISSUE TITLE
AWS Security Hub Integration with Sentinel "Authentication Issue"

### ISSUE THREAD

**Issue by Mandar16161** [Azure/Azure-Sentinel#11165]
**Describe the bug**
Deploying the AWS security hub gives a AAD authentication related issue when test run in the function app.
![image (5)](https://github.com/user-attachments/assets/947cbc9a-fbaf-4c71-993c-5c9cd406e81e)

**To Reproduce**

**Expected behavior**
The steps are followed as mentioned in the document, the managed identity should have authenticated but its failing while runing.

**Screenshots**
![image](https://github.com/user-attachments/assets/d3c3ceca-71c5-4c7e-a893-b420ec49da97)

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Browser chrome

**Smartphone (please complete the following information):**

**Additional context**

**Comment by v-rusraut** [comment]
Hi @Mandar16161,
We have been able to reproduce this issue and found that the error occurred due to an incorrect ClientID value. Please check the attached document to add the correct ClientID value in the function app configuration. 

[FunctionsAppConfiguration.docx](https://github.com/user-attachments/files/17131792/FunctionsAppConfiguration.docx)
Please let us know if your issue has been resolved or not
Thanks

**Comment by v-rusraut** [comment]
Hi @Mandar16161,
We are waiting for your response.

**Comment by Mandar16161** [comment]
Hi @v-rusraut i am not able to access the document shared by you.

**Comment by v-rusraut** [comment]
Hi @Mandar16161,
Please refer below steps 

•	Open Azure portal and search ‘Microsoft Entra ID’
![image](https://github.com/user-attachments/assets/e6529eff-8423-4a91-938e-90c36e7509ea)
 

•	Click on Enterprise applications
![image](https://github.com/user-attachments/assets/f31205b9-f7b0-4b2e-9277-4f5234b7af17)


 




•	Type your function app name in search textbox and copy Application ID
![image](https://github.com/user-attachments/assets/9a5f5782-3583-41df-bf4e-68eae0967830)


 

•	Open function app -> Settings - > Environment variables - > App settings ->
Click on Client ID and paste application ID and click on Apply.
 
![image](https://github.com/user-attachments/assets/37eba21c-6394-4c56-a791-48f7368c5d9a)




•	After that restart the function app
![image](https://github.com/user-attachments/assets/ea6bb551-802a-4a9a-aa86-9b3aa8160050)

**Comment by v-rusraut** [comment]
Hi @Mandar16161,
We are waiting for your response.

**Comment by Mandar16161** [comment]
Hi @v-rusraut,
I have provided the the solution you provided to the team waiting for their response ,will keep you posted.

**Comment by Mandar16161** [comment]
@v-rusraut please let me know did you provide any RBAC role the managed identity during the deployment?

**Comment by v-rusraut** [comment]
Hi @Mandar16161, not provided any role while deployment, what error you are getting now ?

**Comment by Mandar16161** [comment]
Same issue we had the client id in place already .

**Comment by Mandar16161** [comment]
![image](https://github.com/user-attachments/assets/2fa50317-986d-4046-95f4-20fe170f5d42)
this is the error

**Comment by Mandar16161** [comment]
![image](https://github.com/user-attachments/assets/5ff910dd-ea62-44e2-a65c-05ea6b4a3da4)

**Comment by rcscoggin** [comment]
We having same exact issue, ClientID is correct, have verified it multiple times.

**Comment by rcscoggin** [comment]
![2024-10-02_11-58-02](https://github.com/user-attachments/assets/28765b7d-ce85-4e79-9d13-c1cd9600f3e4)

**Comment by Mandar16161** [comment]
Hi @rcscoggin at line no 75 add 
credential = ManagedIdentityCredential( client_id=client_id) it should resolve the issue, also there are numerous other errors once you solve this one the code needs to be rechecked by the developer.

**Comment by v-rusraut** [comment]
Hi @Mandar16161, which new error you are getting now?

**Comment by Mandar16161** [comment]
Hi @v-rusraut 
Last changes made on code  Audience='https://sts.amazonaws.com/' on line number 155 of visual studio but not ran successfully.

**Comment by v-rusraut** [comment]
Hi @Mandar16161,
Please provide more details about the error.

**Comment by v-rusraut** [comment]
> ![2024-10-02_11-58-02](https://private-user-images.githubusercontent.com/5656876/372936889-28765b7d-ce85-4e79-9d13-c1cd9600f3e4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc5NDcxOTUsIm5iZiI6MTcyNzk0Njg5NSwicGF0aCI6Ii81NjU2ODc2LzM3MjkzNjg4OS0yODc2NWI3ZC1jZTg1LTRlNzktOWQxMy1jMWNkOTYwMGYzZTQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMDNUMDkxNDU1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YmNlMWZkMTRiYjkwNTI5ODQ3NTQyZTcyNmEyZTMyNTc2ZTVlODVlNDc1NmM1NDUwYmQ1NjFlN2UxNzlkOTNlNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.zboqij9mkbWxmByNoU3m6FBFfzKIhk0VkhVV_12qLfk)

Hi @rcscoggin,
Please deploy the function app and share the invocation logs.

**Comment by rcscoggin** [comment]
@v-rusraut I don't have any changes in place nor do I see an update in the code on the repo, just the current branch.  Would probably be better for @Mandar16161 to share his logs with his new error after fixing line 75 then another log after modifying 155.

**Comment by v-rusraut** [comment]
Hi @Mandar16161, 
We are waiting for your response, please provide more details about the current error.

**Comment by Mandar16161** [comment]
Hi we have currently stopped the testing due to too many errors, we are co-ordinating with Microsoft Team to get a resolution on this.

**Comment by v-rusraut** [comment]
Hi @Mandar16161, as you mentioned you are coordinating with the Microsoft Team for this, can we close this issue, or do you still need support?

**Comment by rcscoggin** [comment]
Hi, end user here of the solution.  We still have the original issue, we reached out to Microsoft and they said they do not support this code and to contact authors of the code for resolution.  Would ask you not close issue as we would have to open another.  We are still getting the original error that Mandar16161 is getting. thanks.

**Comment by v-rusraut** [comment]
Hi @rcscoggin, we received a response from @sreedharande indicating that the error you’re encountering is due to incorrect AWS side configuration. Please refer to the following section of the README file for AWS side configuration  : https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/AWS-SecurityHubFindings/README.md#aws
![Image](https://github.com/user-attachments/assets/c6ca492e-f63d-4ffc-bb4e-55e3d3c164c9)

**Comment by rcscoggin** [comment]
@v-rusraut right, we are not a new user of this and have deployed the connector over the past years many times.  We have triple checked the AWS settings.  What we are NOT seeing are any entries in the signin logs for the managed identity indicating the code never attempts to properly authenticate.  We had troubleshooted to the same issue that the ManagedIdentityCredential call may be suspect.

Please, confirm on your side that the managed identity would have some entry in the signin logs for a failed attempt since that would happen BEFORE it would attempt to use the AWS api hence resulting in some evidence of the Azure authentication code having been able to at least attempt an authentication.

Thanks,
Rodger

**Comment by v-rusraut** [comment]
Hi @sreedharande , please help on this issue.

**Comment by Mandar16161** [comment]
Hi @v-rusraut we are still not able to resolve the issue, @sreedharande your support on this would be appreciated.

**Comment by rcscoggin** [comment]
@v-rusraut  @sreedharande  any updates on this issue?  Still receiving the following error and the identity associated with function app shows no attempts to AuthN.

11/5/2024, 12:40:00 PM

Information

Authenticating to Azure AD.

11/5/2024, 12:40:00 PM

Information

ManagedIdentityCredential will use App Service managed identity

11/5/2024, 12:40:00 PM

Information

No environment configuration found.

11/5/2024, 12:40:00 PM

Information

ManagedIdentityCredential will use App Service managed identity

11/5/2024, 12:40:00 PM

Information

Request URL: 'http://localhost:8081/msi/token?api-version=REDACTED&resource=REDACTED' Request method: 'GET' Request headers: 'X-IDENTITY-HEADER': 'REDACTED' 'User-Agent': 'azsdk-python-identity/1.15.0 Python/3.9.17 (Linux-5.10.102.2-microsoft-standard-x86_64-with-glibc2.28)' No body was attached to the request

11/5/2024, 12:40:10 PM

Information

Response status: 500 Response headers: 'Content-Type': 'application/json; charset=utf-8' 'Date': 'Tue, 05 Nov 2024 17:40:10 GMT' 'Server': 'Kestrel' 'Transfer-Encoding': 'chunked' 'X-CORRELATION-ID': 'REDACTED'

**Comment by v-sudkharat** [comment]
@sreedharande, Your inputs will really helpful for this issue. Thanks!

**Comment by kausd1** [comment]
@Mandar16161 @rcscoggin 

I reproduced your issue.

Option 1 which is the default option being referred here I would suggest using the option 2 which works well.

As in option 1 - AppId  can not use Managed Service Identity (MSI) as audience in token as it is unsupported. MSI should not be set as audience as it does not accept tokens.

In Option 2 - on the AWS side we use audience when setting up the IDP as the URI of the new application we register.

![Image](https://github.com/user-attachments/assets/b1973b13-937b-4e0e-bfe3-689607c456ae)

On the client ID(Azure function) in the environment variable use the same URI.

https://blog.identitydigest.com/azuread-access-aws/

No changes I had to make to the code.
